src/games/chlcc/titlemenu.cpp

```diff
@@ -1,231 +1,429 @@
+      SecondaryItemHighlightSprite,
+      glm::vec2(SecondaryItemX, ItemConfigY));
+  Config->OnClickHandler = secondaryOnClick;
+  Config->isSubButton = true;
+  Config->LineDecoration = LineSprites[2];
+  Config->LineY = SecondaryMenuSystemConfigY;
+  SystemItems->Add(Config, FocusDirection::Vertical);
+
+  // System Save secondary System menu button
+  SystemSave = new TitleButton(13, MenuEntriesSprites[13], MenuEntriesHSprites[13],
+      SecondaryItemHighlightSprite, 
+      glm::vec2(SecondaryItemX, ItemSystemSaveY));
+  SystemSave->OnClickHandler = secondaryOnClick;
+  SystemSave->isSubButton = true;
+  SystemSave->LineDecoration = LineSprites[3];
+  SystemSave->LineY = SecondaryMenuSystemSaveY;
+  SystemItems->Add(SystemSave, FocusDirection::Vertical);
+}
+
+void TitleMenu::Show() {
+  if (State != Shown) {
+    State = Shown;
+    if (UI::FocusedMenu != 0) {
+      LastFocusedMenu = UI::FocusedMenu;
+      LastFocusedMenu->IsFocused = false;
+    }
+    IsFocused = true;
+    UI::FocusedMenu = this;
+    if (PressToStartAnimation.State == AS_Stopped) {
+      PressToStartAnimation.StartIn();
+      SpinningCircleAnimation.StartIn();
+    }
+  }
+}
+void TitleMenu::Hide() {
+  if (State != Hidden) {
+    State = Hidden;
+    MainItems->Hide();
+    if (LastFocusedMenu != 0) {
+      UI::FocusedMenu = LastFocusedMenu;
+      LastFocusedMenu->IsFocused = true;
+    } else {
+      UI::FocusedMenu = 0;
+    }
+    IsFocused = false;
+  }
+}
+
+void TitleMenu::Update(float dt) {
+  PressToStartAnimation.Update(dt);
+  SpinningCircleAnimation.Update(dt);
+  PrimaryFadeAnimation.Update(dt);
+  SecondaryFadeAnimation.Update(dt);
+
+  if (GetFlag(SF_TITLEMODE)) {
+    Show();
+  } else {
+    ItemsFadeInAnimation.StartOut();
+    SecondaryItemsFadeInAnimation.StartOut();
+    Hide();
+  }
+
+  if (State == Shown && IsFocused) {
+    MainItems->Opacity =
+        glm::smoothstep(0.0f, 1.0f, PrimaryFadeAnimation.Progress);
+    MainItems->Update(dt);
+    LoadItems->Opacity =
+        glm::smoothstep(0.0f, 1.0f, SecondaryFadeAnimation.Progress);
+    LoadItems->Update(dt);
+    ExtraItems->Opacity =
+        glm::smoothstep(0.0f, 1.0f, SecondaryFadeAnimation.Progress);
+    ExtraItems->Update(dt);
+    SystemItems->Opacity =
+        glm::smoothstep(0.0f, 1.0f, SecondaryFadeAnimation.Progress);
+    SystemItems->Update(dt);
+    
+    switch (ScrWork[SW_TITLEDISPCT]) {
+      case 0: {
+        // When returning to title menu from loading a game we need to hide the
+        // load sub-menu
+        if (LoadItems->IsShown) {
+          SecondaryFadeAnimation.StartOut();
+          MainItems->HasFocus = true;
+          LoadItems->Hide();
+        }
+      } break;
+      case 3: {  // Main Menu Fade In
+        if (!MainItems->IsShown && ScrWork[SW_TITLECT] == 0) {
+          MainItems->Show();
+          MainItems->Opacity = 0.0f;
+          PrimaryFadeAnimation.DurationIn = 0.3f;
+          PrimaryFadeAnimation.DurationOut = 0.3f;
```

Comment:
A nitpick really and I know I did the same thing in my MO6TW menu and forgot about that (oops), but these kinds of values should be taken from the profile, just fyi